samcli/lib/build/app_builder.py

```diff
@@ -7,15 +7,14 @@
 import logging
 import pathlib
 from typing import List, Optional, Dict, cast, NamedTuple
-
 import docker
 import docker.errors
 from aws_lambda_builders import (
     RPC_PROTOCOL_VERSION as lambda_builders_protocol_version,
 )
 from aws_lambda_builders.builder import LambdaBuilder
 from aws_lambda_builders.exceptions import LambdaBuilderError
-
+from samcli.commands.build.utils import prompt_user_to_enable_mount_with_write
```

Comment:
Yes, we could do it in the `build_context`. Currently we don't prompt click until we know `_mount_with_write` is a must-have. I might be wrong - if move it to `command`, we will need to introduce the functionality in `workflow_config.get_workflow_config` to determine if `_mount_with_write` is needed. Doing so does move the UX part out of the `lib`, but not separate those components ideally.

Reply:
I don't have anything against prompting but it shouldn't be done within the lib. If the customer is passing in "mount with write", we can check that in the `samcli/commands/build` module. It doesn't need to be tied to a specific workflow (that will just cause more issues in the long run).

Reply:
I don't have anything against prompting but it shouldn't be done within the lib. If the customer is passing in "mount with write", we can check that in the samcli/commands/build module. It doesn't need to be tied to a specific workflow (that will just cause more issues in the long run).

Reply:
Moved prompting to `samcli.build.build_context`. I import `samcli.lib.build.work_config` and assume it's ok since there are already imports from `samlic.lib.build` to `samcli.build.build_context`